src/components/campaigns/DonorsAndDonations.tsx

```diff
@@ -47,8 +47,11 @@ export default function DonorsAndDonations({
   const { t, i18n } = useTranslation()
   const [donationsToShow, setDonationsToShow] = useState<CampaignDonation[] | undefined>()
   const [all, setAll] = useState<boolean>(false)
+  const shownDonationsNumber = 1
   useEffect(() => {
-    all ? setDonationsToShow(donations) : setDonationsToShow(donations?.slice(0, 1))
+    all
```

Comment:
It's gonna be better if we use `useMemo` instead of `useState + useEffect`

```tsx
const shownDonationsNumber = 5
const donationsToShow = useMemo(() => {
  if (all) {
    return donations
  }
  return donations?.slice(0, shownDonationsNumber)
}, [donations, all])
```